Documentation/ceph-upgrade.md

```diff
@@ -48,116 +49,33 @@ With this upgrade guide, there are a few notes to consider:
-
-* Skip this section
-* Set the CSI version to 2.x as described below in the [CSI Version](#csi-version) section.
-
-To continue with the Ceph-CSI v3.0 driver that is recommended with v1.4:
-
-1. List all the volumesnapshots created
-
-```bash
-kubectl get volumesnapshot
-NAME               AGE
-rbd-pvc-snapshot   22s
-```
-
-2. Delete all volumesnapshots
-
-```bash
-kubectl delete volumesnapshot rbd-pvc-snapshot
-volumesnapshot.snapshot.storage.k8s.io "rbd-pvc-snapshot" deleted
-```
-
-3. List all volumesnapshotclasses created
-
-```bash
-kubectl get volumesnapshotclass
-NAME                      AGE
-csi-rbdplugin-snapclass   86s
-```
-
-4. Delete all volumesnapshotclasses
-
-```bash
-kubectl delete volumesnapshotclass csi-rbdplugin-snapclass
-volumesnapshotclass.snapshot.storage.k8s.io "csi-rbdplugin-snapclass" deleted
-```
-
-*Note:* The underlying snapshots on the storage system will be deleted by ceph-csi
-
-#### Delete the Alpha CRDs
-
-As we are updating the snapshot resources from `Alpha` to `Beta` we need to
-delete the old `alphav1` snapshot CRD created by external-snapshotter sidecar container
-
-Check if we have any `v1alpha1` CRD created in our kubernetes cluster
-
-```console
-kubectl get crd volumesnapshotclasses.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-kubectl get crd volumesnapshotcontents.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-kubectl get crd volumesnapshots.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-```
-
-As we have `v1alpha1` CRD created in our kubernetes cluster, we need to delete
-the Alpha CRD
-
-```console
-kubectl delete crd volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
-
-customresourcedefinition.apiextensions.k8s.io "volumesnapshotclasses.snapshot.storage.k8s.io" deleted
-customresourcedefinition.apiextensions.k8s.io "volumesnapshotcontents.snapshot.storage.k8s.io" deleted
-customresourcedefinition.apiextensions.k8s.io "volumesnapshots.snapshot.storage.k8s.io" deleted
-```
-
-Finally, if you desire to use the beta snapshots, check that the [prerequisites](ceph-csi-snapshot.md#prerequisites) are met.
-
+TODO: are there breaking changes?
 
 ## Upgrading the Rook-Ceph Operator
 
 ## Patch Release Upgrades
 
 Unless otherwise noted due to extenuating requirements, upgrades from one patch release of Rook to
-another are as simple as updating the image of the Rook operator. For example, when Rook v1.4.1 is
-released, the process of updating from v1.4.0 is as simple as running the following:
+another are as simple as updating the image of the Rook operator. For example, when Rook v1.5.1 is
+released, the process of updating from v1.5.0 is as simple as running the following:
 
 ```console
-kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.4.1
```

Comment:
With patch releases, we will also now need to recommend applying the latest crds.yaml. It may not be needed every time, but how about if we document it now as a standard step for patch releases?

Reply:
Good idea.